src/main/java/roomescape/user/service/ReservationService.java

```diff
@@ -0,0 +1,50 @@
+package roomescape.user.service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import roomescape.admin.service.ReservationTimeService;
+import roomescape.exception.DataNotFoundException;
+import roomescape.user.domain.Reservation;
+import roomescape.user.repository.reservation.ReservationRepository;
+
+@Service
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeService reservationTimeService;
+
+    public ReservationService(
+            @Qualifier("h2ReservationRepository") final ReservationRepository reservationRepository,
+            final ReservationTimeService reservationTimeService
+    ) {
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeService = reservationTimeService;
+    }
+
+    public Long save(final String name, final LocalDate date, final Long time_id) {
+        final Reservation reservation = new Reservation(name, date, reservationTimeService.getById(time_id));
+
+        return reservationRepository.save(reservation);
+    }
+
+    public Reservation getById(final Long id) {
+        return reservationRepository.findById(id)
+                .orElseThrow(() -> new DataNotFoundException("해당 예약 데이터가 존재하지 않습니다. id = " + id));
+    }
+
+    public List<Reservation> findAll() {
+        return reservationRepository.findAll();
+    }
+
+    public void deleteById(final Long id) {
+        final Optional<Reservation> found = reservationRepository.findById(id);
+
+        if (found.isEmpty()) {
+            throw new DataNotFoundException("해당 예약 데이터가 존재하지 않습니다. id = " + id);
+        }
+        reservationRepository.delete(found.get());
```

Comment:
`orElseThrow` 를 사용하면 아래와 같이 개선할 수 있습니다.

```suggestion
        Reservation found = reservationRepository.findById(id)
                .orElseThrow(() -> new DataNotFoundException("해당 예약 데이터가 존재하지 않습니다. id = " + id));

        reservationRepository.delete(found);
```